### PR TITLE
Wds evolution v2

### PR DESCRIPTION
**Pull request by GregHanson**
**Please provide a description of this PR:**

control plane implementation for https://github.com/istio/ztunnel/pull/529

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ X ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pkg/workloadapi/workload.proto [hunk 1]
```diff
@@ -74,8 +74,22 @@ message Service {
   repeated string subject_alt_names = 6;
 }
 
+// Workload represents a workload - an endpoint (or collection behind a hostname).
+// The xds primary key is "uid" as defined on the workload below.
+// Secondary (alias) keys are the unique `network/IP` pairs that the workload can be reached at.
```
Comment (howardjohn) [inline pkg/workloadapi/workload.proto:79]: I think we need to actually handle the alias by putting them all in the discovery.Resource object. ztunnel also needs to handle this, so if it requests "ip1" and gets back "Key=uid,aliases=[ip1]" it fetches that

Comment (GregHanson) [reply pkg/workloadapi/workload.proto:79]: @howardjohn I added aliases to the discovery.Resource object returned in `pilot/pkg/xds/workload.go`

File: pilot/pkg/xds/workload.go [hunk 2]
```diff
@@ -102,6 +102,7 @@ func (e WorkloadGenerator) GenerateDeltas(
 		have.Insert(n)
 		resources = append(resources, &discovery.Resource{
 			Name:     n,
+			Aliases:  addr.Aliases(),
```
Comment (GregHanson) [inline pilot/pkg/xds/workload.go:105]: @howardjohn is this all that is needed on the control plane side?

File: istio.deps [hunk 3]
```diff
@@ -11,6 +11,6 @@
     "name": "ZTUNNEL_REPO_SHA",
     "repoName": "ztunnel",
```
Comment (kdorosh) [inline istio.deps:12]: todo: update proxy repo sha once https://github.com/istio/proxy/pull/4691 merges

File: pilot/pkg/model/service.go [hunk 4]
```diff
@@ -852,6 +852,22 @@ type AddressInfo struct {
...
+		aliases := make([]string, 0, len(addr.Workload.Addresses))
+		network := addr.Workload.Network
+		for _, workloadAddr := range addr.Workload.Addresses {
+			ip, _ := netip.AddrFromSlice(workloadAddr)
+			aliases = append(aliases, network+"/"+ip.String())
+		}
+		return aliases
+	case *workloadapi.Address_Service:
+		return nil
+	}
```
Comment (kdorosh) [inline pilot/pkg/model/service.go:867]: todo: services are primary keyed by `namespace/hostname` but have `network/IP` aliases too

Comment (kdorosh) [inline pilot/pkg/model/service.go:867]: todo: services are primary keyed by `namespace/hostname` but have `network/IP` aliases too

File: pilot/pkg/model/service.go [hunk 5]
```diff
@@ -852,6 +852,22 @@ type AddressInfo struct {
...
 }
 
+func (i AddressInfo) Aliases() []string {
+	switch addr := i.Type.(type) {
+	case *workloadapi.Address_Workload:
+		aliases := make([]string, 0, len(addr.Workload.Addresses))
+		network := addr.Workload.Network
+		for _, workloadAddr := range addr.Workload.Addresses {
+			ip, _ := netip.AddrFromSlice(workloadAddr)
+			aliases = append(aliases, network+"/"+ip.String())
```
Comment (kdorosh) [inline pilot/pkg/model/service.go:862]: prefer `AddressInfo.ResourceName()` to this aliases func?

**Review by kdorosh — COMMENTED** [review]
direction lgtm, clearly still WIP so not leaving super thorough review

File: pilot/pkg/model/service.go (hunk 4 above)
Comment (GregHanson) [reply pilot/pkg/model/service.go]: done

**Review by kdorosh — APPROVED** [review]
lgtm! once proxy PR merges and we update here per: https://github.com/istio/istio/pull/45170/files#r1213322993

File: istio.deps (hunk 3 above)
Comment (GregHanson) [reply istio.deps:12]: done

**Review by kdorosh — APPROVED** [review]
PTAL @howardjohn

**Review by howardjohn — APPROVED** [review]

**Pull request merged**